icaruscode/CRT/CRTUtils/CRTT0MatchAlg.cc

```diff
@@ -793,5 +851,231 @@ namespace icarus{
+
+   int hit_id = 0;
+
+    // ====================== Matching Algorithm ========================== //
+    //  std::vector<std::pair<sbn::crt::CRTHit, double>> t0Candidates;
+    std::vector<match_geometry> t0Candidates;
+
+    //    if (crtHits.size() == 0) continue;
+    // Loop over all the CRT hits
+    for(auto &crtHit : crtHits){
+      // Check if hit is within the allowed t0 range
+      double crtTime = -99999.;  // units are us
+
+     icarus::match_geometry this_candidate = makeNULLmc_geo();
+     if(fIsData){
+      if (fTSMode == 1) {
+	crtTime = ((double)(int)crtHit.ts1_ns) * 1e-3; //+ fTimeCorrection;
+      }//end if (fTSMode == 1)
+      else {
+		crtTime = double(crtHit.ts0_ns - (trigger_timestamp%1000000000))/1e3;
+		if(crtTime<-0.5e6) 	crtTime+=1e6;
+		else if(crtTime>0.5e6) 	crtTime-=1e6;		
+      }//end else
+     }//end if(fIsData)
+     else if(!fIsData){	
+	crtTime = crtHit.ts0_ns/1e3;
+     }//end else if(!fIsData)
+      // If track is stitched then try all hits
+      if (!((crtTime >= t0MinMax.first - 10. && crtTime <= t0MinMax.second + 10.) 
+            || t0MinMax.first == t0MinMax.second)) continue;
+
+      // cut on CRT hit PE value
+      if (crtHit.peshit<fPEcut) continue;
+      if (crtHit.x_err>fMaxUncert) continue;
+      if (crtHit.y_err>fMaxUncert) continue;
+      if (crtHit.z_err>fMaxUncert) continue;
+      if (tpcTrack.Length() < fMinTrackLength) continue;
+
+      TVector3 crtPoint(crtHit.x_pos, crtHit.y_pos, crtHit.z_pos);
+
+      //Calculate Track direction
+      std::pair<TVector3, TVector3> startEndDir;
+      // dirmethod=2 is original algorithm, dirmethod=1 is simple algorithm for which SCE corrections are possible
+      if (fDirMethod==2)  startEndDir = TrackDirectionAverage(tpcTrack, fTrackDirectionFrac);
+      else startEndDir = TrackDirection(detProp, tpcTrack, fTrackDirectionFrac, crtTime, driftDirection);
+      TVector3 startDir = startEndDir.first;
+      TVector3 endDir = startEndDir.second;
+    
+      // Calculate the distance between the crossing point and the CRT hit, SCE corrections are done inside but dropped
+      double startDist = DistOfClosestApproach(detProp, start, startDir, crtHit, driftDirection, crtTime);
+      double endDist = DistOfClosestApproach(detProp, end, endDir, crtHit, driftDirection, crtTime);
+
+      double xshift = driftDirection * crtTime * detProp.DriftVelocity();
+      auto thisstart = start; 
+      thisstart.SetX(start.X()+xshift);
+      auto thisend = end; 
+      thisend.SetX(end.X()+xshift);
+
+      // repeat SCE correction for endpoints
+      if (fSCE->EnableCalSpatialSCE() && fSCEposCorr) {
+	geo::Point_t temppt = {thisstart.X(),thisstart.Y(),thisstart.Z()};
+	geo::TPCID tpcid = fGeometryService->PositionToTPCID(temppt);
+	geo::Vector_t  fPosOffsets = fSCE->GetCalPosOffsets(temppt,tpcid.TPC);
+	thisstart[0] += fPosOffsets.X();
+	thisstart[1] += fPosOffsets.Y();
+	thisstart[2] += fPosOffsets.Z();
```

Comment:
Defining `thisstart` as `Point_t` should allow this to be intuitively written as:
```suggestion
	geo::TPCID tpcid = fGeometryService->PositionToTPCID(thisstart);
	thisstart += fSCE->GetCalPosOffsets(thisstart,tpcid.TPC);
```